virttest/qemu_vm.py

```diff
@@ -1553,6 +1553,10 @@ def sort_key(dev):
             if backend in ['udp', 'tcp_socket']:
                 serial_params['chardev_host'] = host
                 serial_params['chardev_port'] = free_ports[index]
+            prefix = serial_params.get('virtio_port_name_prefix')
+            serial_name = prefix + str(len(self.virtio_ports))\
```

Comment:
@rdkdd I am trying to keep the original design to make it all the same with before, so does not affect other users, here it used virtio port index as the suffix of port name previously: https://github.com/avocado-framework/avocado-vt/commit/9c82ef693d6a7d5f7835b04d76505f945731fd11#diff-3f2317089f10636f453dca6c4f880f6cL567 I am not sure whether other device use prefix besides spicevmc, so I would like to keep everything back to original way instead of assign any number directly to specific device. Thanks.

Reply:
@rdkdd Seems previously qemu_vm.py also requires you define spice-vdagent as the first virtio serial device with prefix, otherwise you will not get `0` for the device. I suppose that I can only change it to a specific way (like only assign 0 to spice-vdagent) only if there is clear requirement provided for all the devices that require a prefix. There are too many different types of serial device, even only spicevmc has different backends: vdagent/smartcard/usbredir. So I guess better keep it, you can control it by configuration(the order of devices) like before.

Reply:
ok sounds good. LGTM